app/inbox/views.py

```diff
@@ -0,0 +1,148 @@
+import json
+from inbox.models import Notification
+from django.contrib.staticfiles.templatetags.staticfiles import static
+from django.template.response import TemplateResponse
+from django.utils.translation import gettext_lazy as _
+from django.contrib.auth import get_user_model
+from django.http import JsonResponse, HttpResponseForbidden
+from django.core.paginator import Paginator
+from django.views.decorators.csrf import csrf_exempt
+
+
+def notifications(request):
+    """Handle all notifications."""
+    if request.method == 'GET':
+        profile = request.user.profile if request.user.is_authenticated and request.user.profile else None
+        if profile is None:
+            return HttpResponseForbidden('Not Allowed')
+        limit = 10
+        page = 1
+        if 'limit' in request.GET:
+            limit = int(request.GET['limit'])
+        if 'page' in request.GET:
+            page = int(request.GET['page'])
+        print(limit, page)
+        all_notifs = Notification.objects.filter(to_user_id=request.user.id).order_by('id')[::-1]
+        params = dict()
+        all_pages = Paginator(all_notifs, limit)
+        if page > 0 and page <= all_pages.num_pages:
+            all_notifications = []
+            for i in all_pages.page(page):
+                new_notif = i.to_standard_dict()
+                new_notif['created_on'] = i.created_on
+                new_notif['modified_on'] = i.modified_on
+                new_notif['username'] = get_user_model().objects.get(id=new_notif['from_user_id']).username
+
+                all_notifications.append(new_notif)
+            params['data'] = all_notifications
+            params['has_next'] = all_pages.page(page).has_next()
+            params['num_notifications'] = all_pages.count
+            params['num_pages'] = all_pages.num_pages
+        else:
+            params['total_pages'] = all_pages.num_pages
+        return JsonResponse(params, status=200, safe=False)
+
+    else:
+        return HttpResponseForbidden('Not Allowed')
+
+
+@csrf_exempt
+def delete_notifications(request):
+    """For deleting a notification"""
+    profile = request.user.profile if request.user.is_authenticated and request.user.profile else None
+    if request.method == 'DELETE' and profile is not None:
+        params = dict()
+        params['success'] = []
+        try:
+            req_body = json.loads(request.body.decode('utf-8'))
+        except:
+            pass
+        if 'delete' in req_body:
+            for i in req_body['delete']:
+                entry = Notification.objects.filter(id=i)
```

Comment:
If we're looking for notifications by ID, we should perform a `.get()` and catch on `Notification.DoesNotExist`